ovirt_imageio/client/_options.py

```diff
@@ -48,3 +50,45 @@ def parse(self):
         args = self._parser.parse_args()
         # XXX Read config file and merge into args.
         return args
+
+
+class SizeValue(int):
+
+    SUFFIXES = {"": 1, "k": KiB, "m": MiB, "g": GiB, "t": TiB}
+
+    def __str__(self):
+        n = int(self)
+        for unit in self.SUFFIXES:
+            if n < KiB:
+                break
+            n //= KiB
+        return f"{n}{unit}"
+
+
+class Size:
+    """
+    Convert and validate size string.
+    """
+
+    def __init__(self, minimum=0, default=None, maximum=None):
+        self.minimum = None if minimum is None else SizeValue(minimum)
```

Comment:
Is minimum `None` allowed? I ask because the value when omitted is `0`, and I guess a negative size does not make sense.

If you support None for consistency, maybe worth a test to ensure that it works and does not break in the future.

Reply:
We can make minimum=None a hard error, since it does not make sense for a size. Sounds better than adding tests for non existing cases.

Reply:
Sounds good, it also would save having to handle the case where `minimum is None` in the methods below :)

Reply:
mininimum=None raises now TypeError, unneeded check removed.